activemq-kahadb-store/src/main/java/org/apache/activemq/store/kahadb/MultiKahaDBTransactionStore.java

```diff
@@ -427,15 +427,16 @@ private void recoverPendingLocalTransactions() throws IOException {
     }
 
     public JournalCommand<?> load(Location location) throws IOException {
-        DataByteArrayInputStream is = new DataByteArrayInputStream(journal.read(location));
-        byte readByte = is.readByte();
-        KahaEntryType type = KahaEntryType.valueOf(readByte);
-        if (type == null) {
-            throw new IOException("Could not load journal record. Invalid location: " + location);
-        }
-        JournalCommand<?> message = (JournalCommand<?>) type.createMessage();
-        message.mergeFramed(is);
-        return message;
+        try(DataByteArrayInputStream is = new DataByteArrayInputStream(journal.read(location))) {
```

Comment:
Do we actually want to close this? We don't close it in regular KahadDB and the javadocs for DataByteArrayInputStream can be used more than once so I am wondering if there will be any unintentional side effects or if this should have always been closed after loading.

Reply:
Looking at it more carefully i don't think it matters either way. The source value returned is a ByteSequence which just wraps a byte array and that is passed to DataByteArrayInputStream. So in this case there's nothing to actually close, it will just be a no-op.